### PR TITLE
feat(VPN): add new resource vpn gateway

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new resource vpn gateway

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add new resource vpn gateway
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/vpn' TESTARGS='-run TestAccGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccGateway_basic
=== PAUSE TestAccGateway_basic
=== CONT  TestAccGateway_basic
--- PASS: TestAccGateway_basic (175.09s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       175.178s
```

### CODE REVIEW

File: docs/resources/vpn_gateway.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  slave_eip {
+    bandwidth_name = var.bandwidth_name2
+    type           = "5_bgp"
+    bandwidth_size = 5
+    charge_mode    = "traffic"
+  }
+}
+```
+
+### Creating a VPN gateway with existing EIPs
```
Comment (hmbig2) [inline docs/resources/vpn_gateway.md]: change this to basic usage

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+			}
+			statusRaw, err := jmespath.Search(`vpn_gateway.status`, deleteGatewayWaitingRespBody)
+			if err != nil {
+				return nil, "ERROR", fmt.Errorf("error parse %s from response body", `vpn_gateway.status`)
+			}
+
+			status := fmt.Sprintf("%v", statusRaw)
+
+			targetStatus := []string{
+				"xxx",
```
Comment (hmbig2) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go]: remove target or set to empty

Comment (Jason-Zhang9309) [reply huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go]: fixed

File: docs/resources/vpn_gateway.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/vpn_gateway.md]: sure

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/vpn' TESTARGS='-run TestAccGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccGateway_basic
=== PAUSE TestAccGateway_basic
=== CONT  TestAccGateway_basic
--- PASS: TestAccGateway_basic (173.14s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       173.344s
```

File: docs/resources/vpn_gateway.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    id = var.eip_id1
+  }
+
+  slave_eip {
+    id = var.eip_id2
+  }
+}
+```
+
+### Creating a VPN gateway with EIPs
```
Comment (ShiChangkuo) [inline docs/resources/vpn_gateway.md]:
```suggestion
### Creating a VPN gateway with creating new EIPs
```

File: docs/resources/vpn_gateway.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  Defaults to **vpc**
+
+  Changing this parameter will create a new resource.
+
+* `flavor` - (Optional, String, ForceNew) The flavor of the VPN gateway. The value can be **V1G** and **V300**.
+  Defaults to **V300**
+
+  Changing this parameter will create a new resource.
+
+* `bgp_asn` - (Optional, Int, ForceNew) The ASN number of BGP. The value ranges from **1** to **4294967295**.
```
Comment (ShiChangkuo) [inline docs/resources/vpn_gateway.md]: as we discussed before, this parameter should be named `asn`?

File: docs/resources/vpn_gateway.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+  The [object](#Gateway_GetResponseEip) structure is documented below.
+
+<a name="Gateway_GetResponseEip"></a>
+The `master_eip` or `slave_eip` block supports:
+
+* `bandwidth_billing_info` - The bandwidth billing info.
+
+* `bandwidth_id` - The bandwidth ID.
+
+* `billing_info` - The public IP billing info.
```
Comment (ShiChangkuo) [inline docs/resources/vpn_gateway.md]: what are the value of `billing_info` and `bandwidth_billing_info`? I think they are meaningless

File: docs/resources/vpn_gateway.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/vpn_gateway.md]: fixed

File: docs/resources/vpn_gateway.md (hunk 4 above)
Comment (Jason-Zhang9309) [reply docs/resources/vpn_gateway.md]: fixed

File: docs/resources/vpn_gateway.md (hunk 5 above)
Comment (Jason-Zhang9309) [reply docs/resources/vpn_gateway.md]: they are removed

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/vpn' TESTARGS='-run TestAccGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccGateway_basic
=== PAUSE TestAccGateway_basic
=== CONT  TestAccGateway_basic
--- PASS: TestAccGateway_basic (181.56s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       181.800s
```

**Pull request merged**